### DISCUSSION TITLE
Need help translating the code into other frameworks.

### DISCUSSION THREAD

**Discussion by visualDust** [visualDust/ml.akasaki.space#152] [General]
#73 gonna translate the [PyTorch](https://pytorch.org) parts.
do we need a [PaddlePaddle](https://www.paddlepaddle.org) version or something else?

**Comment by visualDust** [comment]
Currently @coronaPolvo working on the PyTorch ver. #73

**Comment by AndPuQing** [comment]
Can we use the version function of Docusaurus to maintain different frameworks? [versioning](https://docusaurus.io/docs/versioning)

**Comment by visualDust** [comment]
we are considering using 'tabs'.  take a look at the [example](https://ml.akasaki.space/playground/tabs/)

**Comment by AndSonder** [comment]
Ooobs, I didn't find the discussion here several minutes ago.

**Comment by AndSonder** [comment]
关于迁移的过程中的问题：建议把每个代码块的switch按钮，改为一个全局的按钮
1. 每个代码块都加一个switch按钮的工序有些过于复杂
2. 有些文档中的代码块过多，都加上switch按钮也不太好看
![image](https://user-images.githubusercontent.com/55493212/141644590-186757d3-a09d-4d70-87b8-f0208b462cd9.png)

3. 有些文字中也包含tensorflow和pytorch的描述，人工去寻找每一个并添加switch按钮工作量有些大

我等会会提一个pr，你可以看一看用一个全局按钮的效果。
也不一定所有的文章都是一个全局按钮，可以自由搭配，但是对于pytorch版本的迁移一个一个的小标题工作量确实有些大  @visualDust

**Reply by visualDust** [reply]
please split the codes as if possible. using tabs as a global switcher sounds just like keeping the repo in multiple versions. hard to maintain, and bad(at least not good enough) for reading the raw file.

**Comment by AndSonder** [comment]
I can't find d2lzh_tensorflow2 in docs/ch1p2/[2]multilayer-perceptron-code.mdx. I need to know the detail in this file before writing the PyTorch version. @visualDust
